### ISSUE TITLE
Allow object-curly-newline to permit newlines around single property

### ISSUE THREAD

**Issue by captbaritone** [eslint/eslint#7609]
**What rule do you want to change?**

    object-curly-newline: ["error", { "multiline": true }]

**Does this change cause the rule to produce more or fewer warnings?**

This change would optionally cause fewer warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option, or new configuration added to an existing option.

**Please provide some example code that this change will affect:**

```js
let a = {
    b: "c"
};
```

**What does the rule currently do for this code?**

Warns "Unexpected a line break after this open brace. (object-curly-newline)" on the first line and "Unexpected a line break before this close brace. (object-curly-newline)" on the last line.

**What will the rule do after it's changed?**

Not warn when an object with a single property places that property on its own line.

---

As @platinumazure said on Gitter:

> It seems like the option enforces two rules: No linebreaks around curlies if all properties are on the same line; and required linebreaks around curlies if all properties are not on the same line.
Seems like you want just the second half of that.

**Comment by vitorbal** [comment]
@captbaritone thank you for your issue! I agree this should be possible for the object-curly-newline rule. How do you imagine the schema to look like with this new option/configuration?

One more thing: is this something that you are willing to work on if it gets accepted? The team has a limited throughput at the moment and we are trying to focus on the JSCS compatibility milestone as much as we can. If you are willing to work on a PR for this, there's a higher chance of it being accepted.

**Comment by captbaritone** [comment]
Thanks for your quick response.

How about adding a new object option, parallel to the `multiline` option, called `singleline` with a description as follows:

* `"singleline": true` (default) disallows line breaks if all properties exist on a single line.

This would be backwards compatible with the existing behavior, but setting it to `false` would allow for the behavior I'm looking for.

This would also allow folks to write configurations which would be harmoinous with the `"allowMultiplePropertiesPerLine"` option of [object-property-newline](http://eslint.org/docs/rules/object-property-newline). If they set `"singleline": false` they could write:

```
var obj2 = {
    foo: "foo", bar: "bar", baz: "baz"
};
```
The naming might be able to be improved, I'm open to suggestions.

I'd be happy to implement the change myself if/when we can agree on an schema/behavior.

**Comment by captbaritone** [comment]
@vitorbal (ping)

**Comment by ivantcov** [comment]
Hey @vitorbal, is their any progress on the issue?

**Comment by vitorbal** [comment]
Sorry @captbaritone @ivantcov, looks like i lost track of this!

@eslint/eslint-team this would be a backwards-compatible enhancement to `object-curly-newline` to allow newlines between curlies if there is only one property in the object literal. Here's the proposed schema: https://github.com/eslint/eslint/issues/7609#issuecomment-261014672. Looks like @captbaritone would also be willing to implement this if it gets accepted. Anyone interested in giving some 👍 's?

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
